apps/prairielearn/assets/scripts/instructorAssessmentInstancesClient.tsx

```diff
@@ -401,93 +415,235 @@ onDocumentReady(() => {
-            type="number"
-            name="time_add"
-            aria-label="Time value"
-            style="width: 5em"
-            value="5"
-          />
-          <select class="custom-select time-limit-field" name="time_ref" aria-label="Time unit">
-            <option value="minutes">minutes</option>
-            ${row.time_remaining_sec !== null
-              ? html`<option value="percent">% total limit</option>`
-              : ''}
-          </select>
-        </p>
-        ${row.has_closed_instance
-          ? html`
-              <div class="form-check mb-2 reopen-closed-field">
+  function TimeLimitExplanation({ action }: { action: TimeLimitAction }) {
+    let explanation = '';
+    if (action === 'set_total') {
+      explanation =
+        'Updating the total time limit will set the given amount of time for the assessment based on when the assessment was started.';
+    } else if (action === 'set_rem') {
+      explanation =
+        'Updating the time remaining will set the given amount of time for the assessment based on the current time.';
+    } else if (action === 'set_exact') {
+      explanation = 'This will set the exact closing time for the assessment.';
+    } else if (action === 'add') {
+      explanation = 'This will add the given amount of time to the remaining time limit.';
+    } else if (action === 'subtract') {
+      explanation = 'This will subtract the given amount of time from the remaining time limit.';
+    } else if (action === 'remove') {
+      explanation = 'This will remove the time limit and the assessment will remain open.';
+    } else if (action === 'expire') {
+      explanation = 'This will expire the time limit and the assessment will be closed.';
+    }
+    return <small class="form-text text-muted">{explanation}</small>;
+  }
+
+  function TimeLimitEditPopover({
+    row,
+  }: {
+    row: {
+      action: string;
+      assessment_instance_id: number;
+      date: string;
+      has_closed_instance: boolean;
+      has_open_instance: boolean;
+      total_time: string;
+      total_time_sec: number;
+      time_remaining: string;
+      time_remaining_sec: number;
+      open: boolean;
+    };
+  }) {
+    const [form, setForm] = useState({
+      action: 'add' as TimeLimitAction,
+      time_add: 5,
+      date: Temporal.Now.zonedDateTimeISO(timezone).toPlainDateTime().toString().slice(0, 16),
+      reopen_closed: false,
+      reopen_without_limit: true,
+    });
+    const showTimeLimitOptions =
+      row.action === 'set_time_limit_all' || row.open || !form.reopen_without_limit;
+
+    function updateFormState<T extends keyof typeof form>(key: T, value: (typeof form)[T]) {
+      setForm({
+        ...form,
+        [key]: value,
+      });
+    }
+
+    function proposedClosingTime() {
+      const startDate = row.date;
+      const totalTime = Math.round(row.total_time_sec);
+
+      let temporalDate = Temporal.Instant.from(startDate).toZonedDateTimeISO(timezone);
```

Comment:
```suggestion
      const totalTime = Math.round(row.total_time_sec);

      let temporalDate = Temporal.Instant.from(row.date).toZonedDateTimeISO(timezone);
```

After applying this, you might also consider renaming `temporalDate` to `startDate`. "Temporal" in the name doesn't mean much other than indicating the variable's type, "start date" is more useful.